storybook/pagesmodel.cpp

```diff
@@ -0,0 +1,90 @@
+#include "pagesmodel.h"
+
+#include <QRegularExpression>
+#include <QDir>
+
+namespace {
+const auto categoryFirst QStringLiteral("_");
+const auto categoryUncategorized QStringLiteral("Uncategorized");
+}
+
+PagesModel::PagesModel(const QString &path, QObject *parent)
+    : QAbstractListModel{parent}
+{
+    QDir dir(path);
+    dir.setFilter(QDir::Files);
+
+    static QRegularExpression fileNameRegex(
+                QRegularExpression::anchoredPattern("(.*)Page\\.qml"));
+    static QRegularExpression categoryRegex(
+                "^//(\\s)*category:(.+)$", QRegularExpression::MultilineOption);
+
+    const QFileInfoList files = dir.entryInfoList();
```

Comment:
This is a blocking call, better use a `QDirIterator`

Reply:
Same as above, not worth to try to optimize it prematurely. Tested on 8K files:

- Linux
  - `QDir::entryInfoList`: 3,5ms
  - `QDirIterator`: 9ms (almost 3x slower (!))
- Windows
    - `QDir::entryInfoList`: 17ms
    - `QDirIterator`: 13ms (slightly faster)
    
I hope this clears up any doubts.

Reply:
Fair enough, thanks for the analysis!